torchvision/io/video_reader.py

```diff
@@ -251,7 +251,12 @@ def get_metadata(self) -> Dict[str, Any]:
                         rate_n = "framerate"
                     metadata[stream.type] = {rate_n: [], "duration": []}
 
-                rate = stream.average_rate if stream.average_rate is not None else stream.sample_rate
+                    rate = (
+                        stream.average_rate
+                        if hasattr(stream, "average_rate")
+                        and stream.average_rate is not None
+                        else stream.sample_rate
+                    )
```

Comment:
Thanks for the PR, I think this should be equivalent to

```suggestion
                    rate = getattr(stream, "average_rate", None) or stream.sample_rate
```

Reply:
Yes, thanks, that's even better!